From 642f597489e6f644d4bd9a0c267e864cabead024 Mon Sep 17 00:00:00 2001
From: Samuel Georges <sam@daftspunk.com>
Date: Sat, 30 Jan 2021 11:47:39 +1100
Subject: [PATCH] Invalidate the session ID to prevent reuse

1. Good logs in
2. Bad captures Good's session cookie
3. Good logs out
4. Session cookie no longer works
5. Good logs in a second time
6. ORIGINAL session cookie works (Bad is also signed in)
---
 src/Auth/Manager.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Auth/Manager.php b/src/Auth/Manager.php
index 1bb65baf6..a8dc26a28 100644
--- a/src/Auth/Manager.php
+++ b/src/Auth/Manager.php
@@ -686,7 +686,7 @@ public function logout()
 
         $this->user = null;
 
-        Session::flush();
+        Session::invalidate();
         Cookie::queue(Cookie::forget($this->sessionKey));
     }
 
